staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -31,12 +31,28 @@ import (
 // the config has no custom TLS options, http.DefaultTransport is returned.
 type tlsTransportCache struct {
 	mu         sync.Mutex
-	transports map[string]*http.Transport
+	transports map[tlsCacheKey]*http.Transport
 }
 
 const idleConnsPerHost = 25
 
-var tlsCache = &tlsTransportCache{transports: make(map[string]*http.Transport)}
+var tlsCache = &tlsTransportCache{transports: make(map[tlsCacheKey]*http.Transport)}
+
+type tlsCacheKey struct {
+	insecure   bool
+	caData     string
+	certData   string
+	keyData    string
+	serverName string
+}
+
+func (t tlsCacheKey) String() string {
+	keyText := "<none>"
+	if len(keyData) > 0 {
+		keyText = "<redacted>"
```

Comment:
I was thinking of hashing it at construction time, but this is at least better than what was there.